share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,628 @@
+```
+def adda s = "{s}b"
+```
+
+* Function times4 takes an argument x and applies the function double twice on x
+```
+def times4 x = double (double x)
+```
+
+* Function aveI takes 2 arguments x and y and adds two values and divide the result by 2. This function shows the other was of defining the parameters using lambda operator
+```
+def aveI = \x\y (x+y)/2
+```
+
+* Function aveR takes 2 arguments of type double and divides the result by 2
+```
+def aveR = \x\y (x +. y) /. 2.0
+```
+
+* Function aveR adds 2 values and provides the remainder 
+```
+def aveM = \x\y (x+y) % 2
+```
+
+* Function duplicate concantenates or copies a given string twice
+```
+def duplicate s  ="{s}{s}"
+```
+
+* integerToUnicode function takes the ASCII input and provides its equivalent character
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger function gives the ASCII value of the given character
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+* strlen function returns the length of the given string
+```
+wake -x 'strlen "abcde"'
+```
+
+* Below function takes a string and returns the middle character of the string
+```
+def middle fullstring =
+  def len = if (strlen fullstring % 2 == 0) then (strlen fullstring / 2) else  ((strlen fullstring / 2) + 1)
+  def outputstr = explode fullstring | take len | reverse | head 
+  outputstr
+```
+
+* Below function takes a string and returns the string removing the first and last character
+```
+def dtrunc fullstring =
+  def len = strlen fullstring
+  def outputstr = explode fullstring | take (len-1) | tail | cat
+  outputstr
+```
+
+* Function incfirst takes a string and a character string and replaces the first character of the string with the given character string
+```
+def incfirst fullstring chrstr = explode fullstring | tail | prepend chrstr | cat
+```
+
+* Function reorder takes a string , split the string and then reverses the splitted string and then concantenates both (Ex: hangover -> overhang)
+```
+def reorder fullstring = 
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2) 
+  cat (laststr ++ firststr)                         
+```
+
+* Function dubmid takes a string, duplicates the middle character and gives out the string (Ex: hapy -> happy))
+```
+def dubmid fullstring =
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2)
+  def dupli = laststr | take 1
+  cat (firststr ++ dupli ++ laststr)
+```
+
+* Takes a string and then removes the last character in a string
+```
+wake -x '"been"| explode | take 3| cat'
+```
+
+* Replaces the first charcter with the character "c"
+```
+wake -x '"bad"| explode | drop 1| prepend "c" | cat' 
+```
+### Tuple
+A tuple is collection of items of different types. `get`,`set` and `edit` are the three inbuilt methods that can be used with the tuples.
+* Below example shows the declaration of a tuple. For more information about the tuple look into wake database
```

Comment:
> For more information about the tuple look into wake database

I don't understand where this info is located. Can you add a link to what you're referring to?

Reply:
Sure. I will add a link at the beginning of the tutorial